.github/workflows/changelog-checker.yml

```diff
@@ -23,13 +23,23 @@ jobs:
           fetch-depth: 0 # by default the checkout action doesn't checkout all branches
       - name: Check for changelog entry in diff
         run: |
+          pull_request_base_main=$(expr github.event.pull_request.base.ref = "main")
```

Comment:
Try interpolating this with the github syntax `${{ github.event.pull_request.base.ref }}`. I don't think it works as written (I haven't tested)

Reply:
```suggestion
          pull_request_base_main=$(expr ${{ github.event.pull_request.base.ref }} = "main")
```